release_dashboard/test/widgets/merge_engine_pr_substeps_test.dart

```diff
@@ -0,0 +1,270 @@
+          expect(find.text(MergePrSubsteps.substepTitles[substep]!), findsOneWidget);
+          expect(find.text(MergePrSubsteps.substepSubtitles[substep]!), findsOneWidget);
+        }
+      });
+
+      testWidgets('Open a PR', (WidgetTester tester) async {
+        await tester.pumpWidget(
+          ChangeNotifierProvider(
+            create: (context) => StatusState(conductor: FakeConductor(testState: stateWithEnginePR)),
+            child: MaterialApp(
+              home: Material(
+                child: ListView(
+                  children: [
+                    MergePrSubsteps(nextStep: () {}, repository: Repositories.engine),
+                  ],
+                ),
+              ),
+            ),
+          ),
+        );
+
+        final String newPrLink = getNewPrLink(
+          userName: githubAccount(kEngineMirror),
+          repoName: 'engine',
+          state: stateWithEnginePR,
+        );
+
+        expect(find.text(newPrLink), findsOneWidget);
+        expect(find.byType(UrlButton), findsNWidgets(2));
+      });
+
+      group('Validate post-submit CI', () {
+        for (final String releaseChannel in kBaseReleaseChannels) {
+          testWidgets('Displays the correct $releaseChannel LUCI dashboard URL', (WidgetTester tester) async {
+            stateWithEnginePR.releaseChannel = releaseChannel;
+            await tester.pumpWidget(
+              ChangeNotifierProvider(
+                create: (context) => StatusState(conductor: FakeConductor(testState: stateWithEnginePR)),
+                child: MaterialApp(
+                  home: Material(
+                    child: ListView(
+                      children: [
+                        MergePrSubsteps(nextStep: () {}, repository: Repositories.engine),
+                      ],
+                    ),
+                  ),
+                ),
+              ),
+            );
+
+            expect(find.text(luciConsoleLink(releaseChannel, 'engine')), findsOneWidget);
+          });
+        }
+      });
+
+      testWidgets('Validate post-submit CI displays an error message if release channel is invalid',
+          (WidgetTester tester) async {
+        stateWithEnginePR.releaseChannel = 'master';
+        await tester.pumpWidget(
+          ChangeNotifierProvider(
+            create: (context) => StatusState(conductor: FakeConductor(testState: stateWithEnginePR)),
+            child: MaterialApp(
+              home: Material(
+                child: ListView(
+                  children: [
+                    MergePrSubsteps(nextStep: () {}, repository: Repositories.engine),
+                  ],
+                ),
+              ),
+            ),
+          ),
+        );
+
+        expect(find.text(MergePrSubsteps.releaseChannelInvalidErr), findsOneWidget);
+      });
+    });
+
+    group('Checksteps and continue button logic tests', () {
+      testWidgets('Continue button appears when all substeps are checked', (WidgetTester tester) async {
```

Comment:
This continue button can't be found for some reasons. I have tried a lot of alternatives.

- Offsets y by +100, -100, -300, -500, -700, -900, -1100, -1500, offset x by -200, 200
- Using Column instead of ListView, get rid of ListView completely
- `find.text('Continue')`, `find.byType(ElevatedButton)`
- `dragUntilVisible`, `pumpAndSettle()`

None of of them find the `continue` button.

Reply:
After a few months of debugging, the tests pass if I enlarge the window size with `tester.binding.window.physicalSizeTestValue = const Size(5000, 5000);`

Reply:
oh, TIL you can do that